presto-main/src/test/java/io/prestosql/operator/scalar/timestamp/TestTimestampLegacy.java

```diff
@@ -13,11 +13,41 @@
  */
 package io.prestosql.operator.scalar.timestamp;
 
+import io.prestosql.Session;
+import org.testng.annotations.Test;
+
+import static io.prestosql.spi.type.TimeZoneKey.getTimeZoneKey;
+import static org.assertj.core.api.Assertions.assertThat;
+
 public class TestTimestampLegacy
         extends BaseTestTimestamp
 {
     protected TestTimestampLegacy()
     {
         super(true);
     }
+
+    @Test
+    public void testCastToTimeWithDaylightSavings()
```

Comment:
This could go into BaseTestTimestamp, to cover the new semantics (even though it was not broken)

Reply:
I thought about that, but this is just testing an implementation detail. (New) timestamps are not sensitive to timezones, so it didn't feel right to be testing that.

Reply:
I would prefer to have a common test. The new semantics are immune to session zone & its DST, but it is OK to verify that with a test ("black box").